src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,42 @@
+package seedu.addressbook.data.person;
+
+/**
+ * Generalization of Phone, Email and Address class
+ */
+public class Contact {
+
+    private String value;
+    private boolean isPrivate;
+
+    public Contact() {
```

Comment:
as suggested above, make this constructor do the assignment rather than have setters for the members.